### PR TITLE
initial commit for multitab - pybricks/support#418

### PR DESCRIPTION
**Pull request by afarago**
This is just a first draft and PR for initial commenting and syncing on a possible direction on the multi tab feature.

### CODE REVIEW

**Comment by afarago** [comment]
I have created an initial concept on a feature branch on a fork for pybricks. (I a noob working with github - please just tell me if I am not on the right path).

At this point I have a draft for

- reading the list of proper localstore items
- creating a project tab on the left
- effectively switching and syncing with the localstore on editor data change (user types code in)
- loading-saving active tab

So my problem is that when OpenFileButton fires onFile --> routed to editor/sagas/open -- there is a reference to the monaco editor (I guess this was and architectural decision).
I am already passing the file name along with the EditorOpenAction

- Yet I would need some reference to the Editor instance to open a new "tab" (handle left menu)
- or even have a higher level restructuring creating a new component above the editor and incorporate the editor and the "project tab" to make a proper abstraction. -- yet my React.js knowledge is far below as of today.

Could you set me on the right path to make the next steps?

**Comment by dlech** [comment]
I would say let's focus on just getting the layout right to start with. As you suggested, the `SplitterLayout` should be added at a higher level in `src/app/App.tsx` and a new component should be created for the project browser (start a new `src/browser/` folder). Don't worry about actually making it work, just fill it with fake data for now, then we can see what it looks like.

File: .eslintrc.js [hunk 1]
```diff
@@ -21,10 +21,16 @@ module.exports = {
...
+        '@typescript-eslint/no-unused-vars': ['off', { argsIgnorePattern: '^_' }],
+        'no-unused-vars': ['off', { argsIgnorePattern: '^_' }],
         'sort-imports': ['error', { ignoreDeclarationSort: true }],
         'import/order': ['error', { alphabetize: { order: 'asc' } }],
+        'prettier/prettier': [
+            'error',
+            {
+                endOfLine: 'auto',
+            },
+        ],
```
Comment (dlech) [inline .eslintrc.js:33]: Let's try not to change anything in this file.

**Comment by dlech** [comment]
Hi @afarago, this turned out to be much more complex than I could have imagined. But it is done now in #823. We should be pushing it out to the beta site within a few days.

**Pull request closed without merging**